### PR TITLE
Fix Zephyr include paths

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

**Comment by pablogs9** [comment]
@mergify backport master foxy galactic

**Review by Acuadros95 — APPROVED** [review]
LGTM

Does the new include path work if `CONFIG_LEGACY_INCLUDE_PATH` is on?

**Comment by pablogs9** [comment]
> Does the new include path work if `CONFIG_LEGACY_INCLUDE_PATH` is on?

It is ON by default. In any case, it works in both cases.

**Comment by pablogs9** [comment]
**Do not merge until decisions on Zephyr version support are taken**

**Comment by uLipe** [comment]
@pablogs9 sorry for taking to long to review this.

The namespace of zephyr headers are a thing for releases of versions 3.x and above, and should not affect previous releases that adopt the namespacing on the headers.

File: modules/libmicroros/microros_transports/serial/microros_transports.c [hunk 1]
```diff
@@ -1,20 +1,27 @@
 #include <uxr/client/transport.h>
 
+#include <microros_transports.h>
+
+#if KERNEL_VERSION_MAJOR >= 3
+#include <zephyr/zephyr.h>
```
Comment (uLipe) [inline modules/libmicroros/microros_transports/serial/microros_transports.c:6]: zephyr.h is deprecated on version 3.x please use kernel.h instead

File: src/main.c [hunk 2]
```diff
@@ -1,8 +1,16 @@
+#if KERNEL_VERSION_MAJOR >= 3
+#include <zephyr/zephyr.h>
```
Comment (uLipe) [inline src/main.c:2]: zephyr.h is deprecated on version 3.x please use kernel.h instead

**Review by uLipe — CHANGES_REQUESTED** [review]
please check the comments regarding zephyr.h

**Comment by uLipe** [comment]
Just to update you, this also broke the rclc component, I will PR the fix.

**Comment by uLipe** [comment]
Here is it:

https://github.com/ros2/rcutils/pull/383

**Comment by uLipe** [comment]
@pablogs9 @Acuadros95 any updates on this?

File: modules/libmicroros/microros_transports/serial/microros_transports.c (hunk 1 above)
Comment (pablogs9) [reply modules/libmicroros/microros_transports/serial/microros_transports.c:6]: Please apply this change

File: src/main.c (hunk 2 above)
Comment (pablogs9) [reply src/main.c:2]: Same as above

**Comment by uLipe** [comment]
@pablogs9 unfortunately I don't have permission to apply here,  I'll dupplicate the PR

**Comment by pablogs9** [comment]
Closing in favor of https://github.com/micro-ROS/micro_ros_zephyr_module/pull/82

**Pull request closed without merging**